doc/shared/FAQ.rst

```diff
@@ -0,0 +1,333 @@
+
+   Suppose your ODE is :math:`y'=f(t,y)` and you integrate it from :math:`0` to :math:`T` and that you are also interested in computing an integral of the form
+
+   .. math::
+
+      z(t) = int_0^t g(t,y(t)) dt
+
+   for some function :math:`g`. The most efficient way of computing :math:`z` is by appending one additional differential equation to your ODE system:
+
+   .. math::
+
+      z' = g(t,y)
+
+   with initial condition :math:`z(0)=0`, in which case the integral from :math:`0` to :math:`T` is `z(T)`.
+
+   This additional equation is "a pure quadrature equation" and its main characteristic is that the
+   new differential variable :math:`z` does not appear in the right hand side of the extended ODE system. If
+   CVODES is notified of such "pure quadrature equations", it can take advantage of this property and
+   do less work than if it didn't know about them (these variables need not be considered in the
+   nonlinear system solution).
+
+   The main reason for the special treatment of "pure quadrature equations" in CVODES is that such
+   integrals (very often a large number of them) need to be computed for adjoint sensitivity.
+
+
+KINSOL
+------
+
+.. collapse:: How do I reinitialize KINSOL within a C/C++ program?
+
+   Although KINSOL does not provide a reinitialization function, it is possible to reinitialize the
+   solver (meaning reuse a KINSOL object), but only if the problem size remains unchanged. To
+   reinitialize KINSOL, begin by making any necessary changes to the problem definition by calling
+   the appropriate ``KINSet*`` functions (e.g., :c:func:`KINSetSysFunc`). Next, if you would like to use
+   a different linear solver, call the appropriate function, followed by any calls to the
+   corresponding ``KIN*Set*`` functions. Then you can call the ``KINSol`` function to solve the updated
+   nonlinear algebraic system.
+
+
+.. collapse:: Why is the system function being evaluated at points that violate the constraints?
+
+   If you have not supplied a function to compute either :math:`J(u)` (of type :c:type:`KINLsJacFn`) or :math:`J(u) v`
+   (of type :c:type:`KINLsJacTimesVecFn`), then the internal function may be the culprit. The
+   default function used to compute a difference quotient approximation to the Jacobian (direct
+   methods) or Jacobian matrix-vector product (Kylov methods) evaluates the user-supplied system
+   function at a slightly perturbed point, but does not check if that point violates the constraints.
+
+
+Miscellaneous
+-------------
+
+.. collapse:: How do I determine which version of SUNDIALS I have?
+
+   If you still have access to the distribution files, then the SUNDIALS release number is indicated
+   in the top-level ``README.md`` and the corresponding solver versions can be determined by
+   reading the appropriate row of the :ref:`release history <History>` table or from the files, ``sundials/src/<solver>/README.md``. You can also call the functions
```

Comment:
The history link does not work as expected because we have two "history" sections:
* `shared/History.rst` and
* `superbuild/source/developers/History.rst`

The first has the release history, so maybe change `.. _History:` to `.. _ReleaseHistory:` at the top of that file to resolve the conflict and any other cross-references to `History` will need to be updated to `ReleaseHistory`.

Reply:
Fixed. There were no other cross-references to History to change.